### PR TITLE
Add a WARNING rule for tagging permission

### PR DESCRIPTION
**Pull request by wenyez**
*Description of changes:*

This change adds a SGR rule. It is to verify when ``tagging`` exists and ``taggable`` is true, the ``tagging.permission`` is specified. Fails this rule will have a WARNING in SGR results.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/integ/data/schema-no-tagging-permission.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            "insertionOrder": false,
+            "items": {
+                "type": "string"
+            }
+        },
+        "MaxSessionDuration": {
+            "description": "The maximum session duration (in seconds) that you want to set for the specified role. If you do not specify a value for this setting, the default value of one hour is applied. This setting can have a value from 1 hour to 12 hours.\n Anyone who assumes the role from the CLI or API can use the ``DurationSeconds`` API parameter or the ``duration-seconds`` CLI parameter to request a longer session. The ``MaxSessionDuration`` setting determines the maximum duration that can be requested using the ``DurationSeconds`` parameter. If users don't specify a value for the ``DurationSeconds`` parameter, their security credentials are valid for one hour by default. This applies when you use the ``AssumeRole*`` API operations or the ``assume-role*`` CLI operations but does not apply when you use those operations to create a console URL. For more information, see [Using IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html) in the *IAM User Guide*.",
+            "type": "integer"
```
Comment (ammokhov) [inline tests/integ/data/schema-no-tagging-permission.json:76]: you dont need a full schema. Just like that would be enough:
```
{
    "tagging": {
        "taggable": true,
        "tagOnCreate": true,
        "tagUpdatable": true,
        "cloudFormationSystemTags": false,
        "tagProperty": "/properties/Tags"
    }
}
```

File: tests/integ/runner/test_integ_runner.py [hunk 2]
```diff
@@ -244,6 +244,74 @@ def test_exec_compliance_stateless_aws_verifiedpermissions_policy(
...
+                        path="",
+                    )
+                },
+                "ensure_properties_do_not_support_multitype": {
+                    GuardRuleResult(
+                        check_id="COM002",
+                        message="type MUST NOT have combined definition",
+                        path="/properties/AssumeRolePolicyDocument/type",
+                    )
+                },
```
Comment (ammokhov) [inline tests/integ/runner/test_integ_runner.py:284]: not necessary, your test iterate over `non_compliant_rules,warning_rules` which are parameters, so you dont check every single check that fails

File: tests/integ/data/schema-no-tagging-permission.json (hunk 1 above)
Comment (wenyez) [reply tests/integ/data/schema-no-tagging-permission.json]: Simplified the json file to include only permission related stuff

File: tests/integ/runner/test_integ_runner.py (hunk 2 above)
Comment (wenyez) [reply tests/integ/runner/test_integ_runner.py]: Updated to only put the TAG012 now~

**Review by ammokhov — APPROVED** [review]

**Review by zkaiweya — APPROVED** [review]

**Pull request merged**